cli/src/diff_util.rs

```diff
@@ -1135,28 +1151,57 @@ pub fn show_diff_summary(
     formatter: &mut dyn Formatter,
     mut tree_diff: TreeDiffStream,
     path_converter: &RepoPathUiConverter,
+    copy_records: &CopyRecords,
+    to_tree: &MergedTree,
 ) -> io::Result<()> {
+    let copied_sources: HashSet<&RepoPath> = copy_records
+        .records()
+        .map(|record| record.source.as_ref())
+        .collect();
+
     async {
         while let Some(TreeDiffEntry {
-            source: _, // TODO handle copy tracking
-            target: repo_path,
+            source: before_path,
+            target: after_path,
             value: diff,
         }) = tree_diff.next().await
         {
             let (before, after) = diff.unwrap();
-            let ui_path = path_converter.format_file_path(&repo_path);
-            if before.is_present() && after.is_present() {
-                writeln!(formatter.labeled("modified"), "M {ui_path}")?;
-            } else if before.is_absent() {
-                writeln!(formatter.labeled("added"), "A {ui_path}")?;
+            let before_ui_path = path_converter.format_file_path(&before_path);
```

Comment:
nit: move into if-block below so we avoid calculating it when it won't be needed (as it won't be most of time)

Reply:
Done